kolibri/content/models.py

```diff
@@ -235,24 +188,53 @@ def get_download_url(self):
         Return the download url.
         """
         new_filename = self.get_download_filename()
-        return reverse('downloadcontent', kwargs={'filename': self.get_filename(), 'new_filename': new_filename})
+        return reverse('downloadcontent', kwargs={'filename': self.local_file.get_filename(), 'new_filename': new_filename})
 
 
 @python_2_unicode_compatible
-class License(ContentDatabaseModel):
+class LocalFile(models.Model):
+    """
+    The bottom layer of the contentDB schema, defines the local state of files on the device storage.
+    """
+    # ID should be the checksum of the file
+    id = UUIDField(primary_key=True)
```

Comment:
Hrm, might be best just to change to a CharField, as we need no other enforcement.